pyproject.toml

```diff
@@ -0,0 +1,43 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "graphene"
+version = "3.3.0"
+authors = [ "Syrus Akbary <me@syrusakbary.com>" ]
+description = "GraphQL Framework for Python"
+readme = "README.md"
+license = "MIT"
+keywords = ["api", "graphql", "protocol", "rest", "relay", "graphene"]
+homepage = "https://graphene-python.org/"
+repository = "https://github.com/graphql-python/graphene"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "Topic :: Software Development :: Libraries",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+]
+
+[tool.poetry.dependencies]
+python = "^3.8"
+graphql-core = ">=3.1,<3.3"
+graphql-relay = ">=3.1,<3.3"
+
+[tool.poetry.group.test.dependencies]
+tox = "*"
+pytest = ">=8,<9"
+pytest-benchmark = ">=4,<5"
+pytest-cov = ">=5,<6"
+pytest-mock = ">=3,<4"
+pytest-asyncio = ">=0.16,<2"
+snapshottest = ">=0.6,<1"
+coveralls = ">=3.3,<5"
```

Comment:
I think you missed ruff here which was added in another part of the setup.py file. Not sure if a separate group for "dev" is required in the same way as it existed in setup.py.

Reply:
I'm moving Ruff to pre-commit hook for lint consistency.

Reply:
Ah ok, though that would depend on developers having pre commit hooks enabled and you wouldn't validate if this has been done within the CI.

I think it can be nice to be able to install everything that is needed within the project so that you can configure the editor to run formatters on save etc.

But this was only a minor comment if there was some oversight.

Reply:
Thank you for suggestion. I have an idea to refactor how lint is run, perhaps not rely on pre-commit as its purpose is just not check but fix issues during commit.